icu4c/source/i18n/unicode/simplenumberformatter.h

```diff
@@ -0,0 +1,331 @@
+
+    /**
+     * Truncates the most significant digits to the given maximum number of integer digits.
+     *
+     * This function immediately mutates the inner value.
+     *
+     * @draft ICU 73
+     */
+    void truncateStart(uint32_t maximumIntegerDigits, UErrorCode& status);
+
+    /**
+     * Pads the beginning of the number with zeros up to the given minimum number of integer digits.
+     *
+     * This setting is applied upon formatting the number. 
+     *
+     * @draft ICU 73
+     */
+    void setMinimumIntegerDigits(uint32_t minimumIntegerDigits, UErrorCode& status);
+
+    /**
+     * Pads the end of the number with zeros up to the given minimum number of fraction digits.
+     *
+     * This setting is applied upon formatting the number.
+     *
+     * @draft ICU 73
+     */
+    void setMinimumFractionDigits(uint32_t minimumFractionDigits, UErrorCode& status);
+
+    /**
+     * Sets the sign of the number: an explicit plus sign, explicit minus sign, or no sign.
+     *
+     * This setting is applied upon formatting the number.
+     *
+     * NOTE: This does not support accounting sign notation.
+     *
+     * @draft ICU 73
+     */
+    void setSign(USimpleNumberSign sign, UErrorCode& status);
+
+    /**
+     * Creates a new, empty SimpleNumber that does not contain a value.
+     * 
+     * NOTE: This number will fail to format; use forInteger to create a SimpleNumber with a value.
+     *
+     * @draft ICU 73
+     */
+    SimpleNumber() = default;
+
+    /**
+     * Destruct this SimpleNumber, cleaning up any memory it might own.
+     *
+     * @draft ICU 73
+     */
+    ~SimpleNumber() {
+        cleanup();
+    }
+
+    /**
+     * SimpleNumber move constructor.
+     *
+     * @draft ICU 73
+     */
+    SimpleNumber(SimpleNumber&& other) U_NOEXCEPT {
+        fData = other.fData;
+        fSign = other.fSign;
+        other.fData = nullptr;
+    }
+
+    /**
+     * SimpleNumber move assignment.
+     *
+     * @draft ICU 73
+     */
+    SimpleNumber& operator=(SimpleNumber&& other) U_NOEXCEPT {
+        cleanup();
+        fData = other.fData;
+        fSign = other.fSign;
+        other.fData = nullptr;
+        return *this;
+    }
+
+  private:
+    SimpleNumber(impl::UFormattedNumberData* data, UErrorCode& status);
+    SimpleNumber(const SimpleNumber&) = delete;
+    SimpleNumber& operator=(const SimpleNumber&) = delete;
+
+    void cleanup();
+
+    impl::UFormattedNumberData* fData = nullptr;
+    USimpleNumberSign fSign = UNUM_SIMPLE_NUMBER_NO_SIGN;
+
+    friend class SimpleNumberFormatter;
+};
+
+
+/**
+ * A special NumberFormatter focused on smaller binary size and memory use.
+ * 
+ * SimpleNumberFormatter is capable of basic number formatting, including grouping separators,
+ * sign display, and rounding. It is not capable of currencies, compact notation, or units.
+ *
+ * This class is immutable and not intended for public subclassing.
```

Comment:
Suggested addition:
This class is movable but not copyable.

Reply:
done